lib/bulk_insert/worker.rb

```diff
@@ -76,14 +77,21 @@ def after_save(&block)
     def save!
       if pending?
         @before_save_callback.(@set) if @before_save_callback
-        compose_insert_query.tap { |query| @connection.execute(query) if query }
+        execute_query
         @after_save_callback.() if @after_save_callback
         @set.clear
       end
 
       self
     end
 
+    def execute_query
+      if query = compose_insert_query
+        result_set = @connection.execute(query)
```

Comment:
`execute` returns the raw result from the adapter, in our case `PG::Result`. There is a long story here, but the **TL;DR** is `PG::Result` has a quasi-memory leak so consider using `exec_query` when possible, which returns an `ActiveRecord::Result`.